packages/x-date-pickers/src/internals/hooks/usePicker/usePicker.types.ts

```diff
@@ -63,9 +63,11 @@ export interface UsePickerResponse<
   TView extends DateOrTimeViewWithMeridiem,
   TError,
 > extends Pick<UsePickerValueResponse<TValue, TError>, 'open' | 'actions' | 'fieldProps'>,
-    Omit<UsePickerViewsResponse<TView>, 'layoutProps' | 'views'> {
+    Pick<UsePickerViewsResponse<TView>, 'shouldRestoreFocus' | 'renderCurrentView'> {
   ownerState: PickerOwnerState;
   providerProps: UsePickerProviderReturnValue;
   layoutProps: UsePickerValueResponse<TValue, TError>['layoutProps'] &
     UsePickerViewsResponse<TView>['layoutProps'];
+  // TODO v8: Remove in https://github.com/mui/mui-x/pull/15671
+  hasUIView: boolean;
```

Comment:
After #15671, the picker hooks (`useDesktopPicker` & co) won't need this variable anymore.